platforms/core-runtime/build-profile/src/main/java/org/gradle/profile/ProfileEventAdapter.java

```diff
@@ -97,15 +97,17 @@ public void afterEvaluate(Project project, ProjectState state) {
     @Override
     public void beforeExecute(TaskIdentity<?> taskIdentity) {
         long now = clock.getCurrentTime();
-        ProjectProfile projectProfile = buildProfile.getProjectProfile(taskIdentity.getProjectPath());
-        projectProfile.getTaskProfile(taskIdentity.getTaskPath()).setStart(now);
+        String projectPath = taskIdentity.getProjectIdentity().getProjectPath().getPath();
+        ProjectProfile projectProfile = buildProfile.getProjectProfile(projectPath);
+        projectProfile.getTaskProfile(taskIdentity.getPath().getPath()).setStart(now);
```

Comment:
❓ I don't think I like the move from `getTaskPath()` to `getPath().getPath()`, I have absolutely no idea what type that's going to be at the end and the double name is hard to read.

Options I prefer:
- Keep `getTaskPath()` for reduced diff noise.
- Add a new `getRawTaskPath()` or `getTaskPathAsString()` or any other name that indicates the nature of the return type.
- Add an `asString()` (or similar) to `Path`, deprecate `getPath()` in favor of that.

Reply:
I'm also not a fan of the double getter call, but this problem is not limited to this getter on the task. Keeping the uses of the `getTaskPath()` getter means using deprecated methods, while we have a better alternative type-wise.

> Add an asString() (or similar) to Path, deprecate getPath() in favor of that.

I like this option the most, and I'd vote for doing it as a follow-up PR and getting feedback from the larger team on it.